### PR TITLE
Created Subintro section

### LINKED ISSUE #13: (SP: 1) Implement subintro section
image

https://oliagrytsak1991.wixsite.com/my-site-2

### PR DESCRIPTION
**Pull request by ValKoDim**
- Added Subintro component and test for it
- Added icon images for each subintro element
Screenshot 2024-06-20 at 14 56 53

Test run:
Screenshot 2024-06-24 at 10 57 51

### CODE REVIEW

File: code/src/components/subintro/Subintro.test.jsx
Comment (mhevyk) [inline code/src/components/subintro/Subintro.test.jsx]: Please format it

File: code/src/components/subintro/Subintro.tsx
Comment (mhevyk) [inline code/src/components/subintro/Subintro.tsx]: Use import alias

File: code/src/components/subintro/Subintro.tsx
Comment (mhevyk) [inline code/src/components/subintro/Subintro.tsx]: Use import aliases

File: code/src/components/subintro/Subintro.tsx
Comment (mhevyk) [inline code/src/components/subintro/Subintro.tsx]: Use default export here

File: code/src/components/subintro/Subintro.tsx
Comment (mhevyk) [inline code/src/components/subintro/Subintro.tsx]: Format this file

File: code/src/components/subintro/subintroData.ts
Comment (mhevyk) [inline code/src/components/subintro/subintroData.ts]: Also format here

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (mhevyk) [inline code/src/components/subintroElement/SubintroElement.tsx]: Use default imports

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (mhevyk) [inline code/src/components/subintroElement/SubintroElement.tsx]: Use separate type like following:

```tsx
type SubintroElement = {
  content: string;
  icon: string;
}

const SubintroElement = ({ content, icon }: SubintroElement) => {
    // ...
}
```

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (mhevyk) [inline code/src/components/subintroElement/SubintroElement.tsx]: Format this document

File: code/src/components/subintro/subintro.scss
Comment (D1nviz) [inline code/src/components/subintro/subintro.scss]: Use css variable for font size, font family and font weight. Look example in AppButton or App typography

File: code/src/components/subintro/Subintro.test.jsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.test.jsx]: Done

Comment (ValKoDim) [reply code/src/components/subintro/Subintro.test.jsx]: Done

File: code/src/components/subintro/Subintro.tsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.tsx]: Done

File: code/src/components/subintro/Subintro.tsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.tsx]: Done

File: code/src/components/subintro/Subintro.tsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.tsx]: Done

File: code/src/components/subintro/Subintro.tsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.tsx]: Done

File: code/src/components/subintro/subintroData.ts
Comment (ValKoDim) [reply code/src/components/subintro/subintroData.ts]: Done

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (ValKoDim) [reply code/src/components/subintroElement/SubintroElement.tsx]: Done

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (ValKoDim) [reply code/src/components/subintroElement/SubintroElement.tsx]: Done

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (ValKoDim) [reply code/src/components/subintroElement/SubintroElement.tsx]: Done

File: code/src/components/subintro/subintro.scss
Comment (ValKoDim) [reply code/src/components/subintro/subintro.scss]: Done

File: code/src/components/subintro/Subintro.test.jsx
Comment (OliaHrytsak) [inline code/src/components/subintro/Subintro.test.jsx]: Please, add alieses there!

File: code/src/components/subintro/subintroData.ts
Comment (OliaHrytsak) [inline code/src/components/subintro/subintroData.ts]: You need to add translation into Ukraine
It should be stored there

image

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (OliaHrytsak) [inline code/src/components/subintroElement/SubintroElement.tsx]: change <p/> to <AppTypography/> component
Look for AppBanner.tsx like example

**Review by OliaHrytsak — CHANGES_REQUESTED** [review]

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (mhevyk) [inline code/src/components/subintroElement/SubintroElement.tsx]: Make new line after it

File: code/src/components/subintro/Subintro.tsx
Comment (mhevyk) [inline code/src/components/subintro/Subintro.tsx]: Was it formatted before?

File: code/src/components/subintro/subintro.scss
Comment (OliaHrytsak) [inline code/src/components/subintro/subintro.scss]: All this styles is inside AppTypography

File: code/src/components/subintro/Subintro.test.jsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.test.jsx]: Done

File: code/src/components/subintro/subintroData.ts
Comment (ValKoDim) [reply code/src/components/subintro/subintroData.ts]: Done

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (ValKoDim) [reply code/src/components/subintroElement/SubintroElement.tsx]: Done

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (ValKoDim) [reply code/src/components/subintroElement/SubintroElement.tsx]: Done

File: code/src/components/subintro/Subintro.tsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.tsx]: Yes, I am using default formatting from VS code.

File: code/src/components/subintro/Subintro.test.tsx
Comment (mhevyk) [inline code/src/components/subintro/Subintro.test.tsx]: Maybe we need to wait until #45 will be merged. Then we should remove these lines

File: code/src/components/subintro/Subintro.test.tsx
Comment (mhevyk) [inline code/src/components/subintro/Subintro.test.tsx]: strange spacing here. Fix it

File: code/src/components/subintro/Subintro.test.tsx
Comment (mhevyk) [inline code/src/components/subintro/Subintro.test.tsx]: I suggest refactoring to `render all subintro elements correctly`

File: code/src/components/subintro/subintro.scss
Comment (mhevyk) [inline code/src/components/subintro/subintro.scss]: Maybe little fix, but maybe use just `0` instead of `0px`

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (mhevyk) [inline code/src/components/subintroElement/SubintroElement.tsx]: We should follow [BEM](https://getbem.com/) methodology for classes

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (mhevyk) [inline code/src/components/subintroElement/SubintroElement.tsx]: Same here

File: code/src/components/subintro/subintro.scss
Comment (OliaHrytsak) [inline code/src/components/subintro/subintro.scss]: &__element - its correct (two underlines)

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (ValKoDim) [reply code/src/components/subintroElement/SubintroElement.tsx]: What is wrong with the naming of this class?

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (ValKoDim) [reply code/src/components/subintroElement/SubintroElement.tsx]: What is wrong with the naming here?

File: code/src/components/subintro/subintro.scss
Comment (OliaHrytsak) [inline code/src/components/subintro/subintro.scss]: Do you really need this  font-weight?

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (mhevyk) [reply code/src/components/subintroElement/SubintroElement.tsx]: We should `spa-subintro__element`

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (mhevyk) [reply code/src/components/subintroElement/SubintroElement.tsx]: Use `spa-subintro__icon`

File: code/src/components/subintro/Subintro.test.tsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.test.tsx]: Yes, i also think it will be better to wait. I will remove those mocks.

File: code/src/shared/modules/application/i18n/uk.json
Comment (OliaHrytsak) [inline code/src/shared/modules/application/i18n/uk.json]: Change into 'Підбір замовлення біля магазину' please

File: code/src/shared/modules/application/i18n/uk.json
Comment (OliaHrytsak) [inline code/src/shared/modules/application/i18n/uk.json]: thirdParagraph

File: code/src/components/subintro/Subintro.constants.ts
Comment (OliaHrytsak) [inline code/src/components/subintro/Subintro.constants.ts]: thirdParagraph

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (ValKoDim) [reply code/src/components/subintroElement/SubintroElement.tsx]: Oh, its double underscore. I got you.

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (ValKoDim) [reply code/src/components/subintroElement/SubintroElement.tsx]: Oh, its double underscore. I got you.

File: code/src/components/subintro/Subintro.test.tsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.test.tsx]: Done

File: code/src/components/subintro/Subintro.test.tsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.test.tsx]: Done

File: code/src/components/subintro/subintro.scss
Comment (ValKoDim) [reply code/src/components/subintro/subintro.scss]: Done

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (ValKoDim) [reply code/src/components/subintroElement/SubintroElement.tsx]: Done

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (ValKoDim) [reply code/src/components/subintroElement/SubintroElement.tsx]: Done

File: code/src/components/subintro/Subintro.tsx
Comment (D1nviz) [inline code/src/components/subintro/Subintro.tsx]: move this mui import to the top of the file, because it's imported from the library. Move style file to the bottom

File: code/src/components/subintro/Subintro.tsx
Comment (D1nviz) [inline code/src/components/subintro/Subintro.tsx]: separate this map into variable and move it before return

File: code/src/components/subintro/Subintro.test.tsx
Comment (D1nviz) [inline code/src/components/subintro/Subintro.test.tsx]: make space after imports

File: code/src/components/subintro/Subintro.test.tsx
Comment (mhevyk) [inline code/src/components/subintro/Subintro.test.tsx]: You can remove these imports now

Comment (dmtrth25) [reply code/src/components/subintro/Subintro.test.tsx]: > You can remove these imports now

I guess he should to try run tests also after rebase with our rule for mock assets and then remove imports

File: code/src/components/subintro/Subintro.tsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.tsx]: > separate this map into variable and move it before return

Its not necessary, because this components renders only Subintro data.

File: code/src/components/subintro/Subintro.tsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.tsx]: Done

File: code/src/components/subintro/Subintro.test.tsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.test.tsx]: Done

File: code/src/components/subintro/Subintro.test.tsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.test.tsx]: Refactored the test. Now it passes locally, without those mocks.

File: code/src/components/subintro/Subintro.test.tsx
Comment (mhevyk) [inline code/src/components/subintro/Subintro.test.tsx]: Double check if your tests work with provider of not

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (mhevyk) [inline code/src/components/subintroElement/SubintroElement.tsx]: We use convention for types. Our props should be called `ComponentName` + `Props`, in your case `SubintroElementProps`

File: code/src/components/subintro/Subintro.test.tsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.test.tsx]: They work only with that provider. Without it it start throwing errors that it needs said provider.

Comment (dmtrth25) [reply code/src/components/subintro/Subintro.test.tsx]: > They work only with that provider. Without it it start throwing errors that it needs said provider.

Try to `rebase` from develop branch. 
It should work without import `IntlProvider` for tests

File: code/src/components/subintroElement/SubintroElement.tsx
Comment (ValKoDim) [reply code/src/components/subintroElement/SubintroElement.tsx]: Done

**Review by OliaHrytsak — APPROVED** [review]

File: code/src/components/subintro/Subintro.test.tsx
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.test.tsx]: I did and it doesn't work without it.

File: code/src/shared/modules/application/i18n/en.json
Comment (dmtrth25) [inline code/src/shared/modules/application/i18n/en.json]: `Welcome` typo

File: code/src/components/subintro/Subintro.test.tsx
Comment (dmtrth25) [reply code/src/components/subintro/Subintro.test.tsx]: > I did and it doesn't work without it.

We should to discuss it
Don't forget to include screenshot running tests result to the description of this PR

File: code/src/shared/modules/application/i18n/en.json
Comment (ValKoDim) [reply code/src/shared/modules/application/i18n/en.json]: Done

File: code/src/components/subintro/Subintro.constants.ts
Comment (ValKoDim) [reply code/src/components/subintro/Subintro.constants.ts]: Done

File: code/src/shared/modules/application/i18n/uk.json
Comment (ValKoDim) [reply code/src/shared/modules/application/i18n/uk.json]: Done

File: code/src/shared/modules/application/i18n/uk.json
Comment (ValKoDim) [reply code/src/shared/modules/application/i18n/uk.json]: Done

File: code/src/shared/modules/application/application.tsx
Comment (mhevyk) [inline code/src/shared/modules/application/application.tsx]: Use prettier here

**Comment by mhevyk** [comment]
Please perform a rebase

File: code/src/shared/modules/application/application.tsx
Comment (ValKoDim) [reply code/src/shared/modules/application/application.tsx]: Done

File: code/src/components/subintro/Subintro.test.tsx
Comment (dmtrth25) [reply code/src/components/subintro/Subintro.test.tsx]: > I did and it doesn't work without it.

Remove `IntlProvider` with `paragraphs` and wrapper for `Subintro` like e.g.

```suggestion
import { render, screen } from "@testing-library/react";
import Subintro from "@/components/subintro/Subintro";
import subintroElements from "@/components/subintro/Subintro.constants";

describe("Subintro section", () => {
  test("renders four subintro elements correctly", () => {
    render(<Subintro />);

    const subintroItems = screen.getAllByTestId("spa-subintro-item");
    expect(subintroItems).toHaveLength(subintroElements.length);
  });
});
```

Comment (ValKoDim) [reply code/src/components/subintro/Subintro.test.tsx]: Yep, that works

**Comment by D1nviz** [comment]
Also, you should move the subintro folder into the layout folder. Rename SubintroElement to SubintroItem and put it into component folder. Don't forget, names of the folders should be in kebab case

File: code/src/layouts/app-banner/messages/en.json
Comment (D1nviz) [inline code/src/layouts/app-banner/messages/en.json]: You should move translations for subintro to separate translations folder for subintro

Comment (ValKoDim) [reply code/src/layouts/app-banner/messages/en.json]: Where should this folder be located? Inside the component folder?

Comment (D1nviz) [reply code/src/layouts/app-banner/messages/en.json]: inside the layout folder. Check #73 PR
image

Comment (ValKoDim) [reply code/src/layouts/app-banner/messages/en.json]: Okay, done

File: code/src/layouts/subintro/Subintro.tsx
Comment (mhevyk) [inline code/src/layouts/subintro/Subintro.tsx]: Use `AppBox`

File: code/src/layouts/subintro/Subintro.tsx
Comment (mhevyk) [inline code/src/layouts/subintro/Subintro.tsx:4]: Add new line before it

File: code/src/layouts/subintro/components/SubintroItem.tsx
Comment (mhevyk) [inline code/src/layouts/subintro/components/SubintroItem.tsx]: Use `AppBox` instead

**Review by D1nviz — APPROVED** [review]

**Review by OliaHrytsak — APPROVED** [review]

**Review by dmtrth25 — APPROVED** [review]

File: code/src/assets/images/subintro/clock-icon.jpg
Comment (D1nviz) [inline code/src/assets/images/subintro/clock-icon.jpg:1]: move subintro images folder inside the homepage folder

File: code/src/layouts/subintro/components/SubintroItem.tsx
Comment (ValKoDim) [reply code/src/layouts/subintro/components/SubintroItem.tsx]: Done

File: code/src/layouts/subintro/Subintro.tsx
Comment (ValKoDim) [reply code/src/layouts/subintro/Subintro.tsx:4]: Done

File: code/src/layouts/subintro/Subintro.tsx
Comment (ValKoDim) [reply code/src/layouts/subintro/Subintro.tsx]: Done

**Review by mhevyk — APPROVED** [review]

**Pull request closed without merging**
